app/code/core/Mage/Customer/Model/Address.php

```diff
@@ -68,6 +68,8 @@ public function setCustomerId($id)
     {
         $this->setParentId($id);
         $this->setData('customer_id', $id);
+        if (empty($this->getOrigData('customer_id')))
+            $this->setOrigData('customer_id', $id);
```

Comment:
I don't get this. Can you explain why?

Reply:
Shouldn't the `setOrigData` call be moved to `getAddressesCollection` method then?

Reply:
Why not.

Reply:
I think `customer_id` should not be added to `$_origData`. Reasons:

1. It is not set in the address model before this PR.
2. It has no relation to saving or not saving of address model.
3. It may have unintended side-effect. For example, I happen to have a custom module to detect changes in address model, this PR will cause an additional entry `customer_id` when an address is created. Not a breaking change but it has no purpose.
4. Note that both `$_data` and `$_origData` are set at `load()`, including EAV models. My interpretation is that `$origData` is based on the values in DB as I have not seen the core adding any other non-DB key-value pair to it. Reference:

https://github.com/OpenMage/magento-lts/blob/cc3eff3ebec46efbcab7ae4c26dd8a754476da18/app/code/core/Mage/Eav/Model/Entity/Abstract.php#L958-L960

Reply:
So, I will remove it.